### PR TITLE
add support for heterogeneous files

### LINKED ISSUE #139: Files with inconsistent image slice sizes break loading
Currently TIFF images with embedded reduced resolution images cause problems since `TiffImages` assumes the images aren't ragged.

Take [DAPI Myog Pax7_CH1.zip](https://github.com/tlnagy/TiffImages.jl/files/13620275/DAPI.Myog.Pax7_CH1.zip) for example, which fails to load:

```julia
julia> TiffImages.load("DAPI Myog Pax7_CH1.tif")
ERROR: EOFError: read end of file
Stacktrace:
  [1] unsafe_read(s::IOStream, p::Ptr{UInt8}, nb::UInt64)
    @ Base ./iostream.jl:428
  [2] unsafe_read
    @ ./io.jl:761 [inlined]
  [3] read!
    @ ./io.jl:779 [inlined]
  [4] read!(file::TiffFile{UInt32, FileIO.Stream{FileIO.DataFormat{:TIFF}, IOStream, String}}, arr::Matrix{ColorTypes.RGB{FixedPointNumbers.N0f8}})
```

However, loading the image planes separately works:

```julia
tf = read(open("DAPI Myog Pax7_CH1.tif"), TiffFile)
ifds = collect(tf)
TiffImages.load!.(Ref(tf), ifds)
size(TiffImages.load(tf, ifds[1:1], nothing))
```
(720, 960)

and 

```julia
size(TiffImages.load(tf, ifds[2:2], nothing))
```
(120, 160).

It appears as though we are handling everything just fine, it's just that we should detect when images are "ragged", i.e. don't have the same XY sizes and load those images as the currently unimplemented [`StridedTaggedImage`](https://github.com/tlnagy/TiffImages.jl/blob/bda25ffe31737a5ab93e6f470ea933aea3961384/src/types/strided.jl#L1-L5)

### PR DESCRIPTION
**Pull request by chrstphrbrns**
Fixes #139

### CODE REVIEW

File: src/layout.jl [hunk 1]
```diff
@@ -9,10 +9,18 @@ predictor(ifd::IFD) = Int(getdata(ifd, PREDICTOR, 0))
...
 is_irregular_bps(ifd::IFD) = bitspersample(ifd) != sizeof(rawtype(ifd)) * 8
 is_complicated(ifd::IFD) = !iscontiguous(ifd) || compression(ifd) != COMPRESSION_NONE || is_irregular_bps(ifd) == true || predictor(ifd) > 1
 
+# returns true if all slices have the same size and color type
+function is_homogeneous(ifds::Vector{<:IFD})
+    return all(map(==(nrows(first(ifds))), nrows.(ifds))) &&
+        all(map(==(ncols(first(ifds))), ncols.(ifds))) &&
+        all(map(==(colortype(first(ifds))), colortype.(ifds)))
+end
+
```
Comment (tlnagy) [inline src/layout.jl:23]: Is this sufficient to determine homogeneity? I guess we can try with this and add more checks if they end up causing issues.

**Review by tlnagy — APPROVED** [review]
This looks great!

**Comment by tlnagy** [comment]
Awesome! Is there anything else you want to add to the release? I see that you bumped the version to 1.0, but I would like to fix #135 prior tagging v1.0 so I might make a new PR changing the release to v0.10.0

**Comment by chrstphrbrns** [comment]
I have a PR for #100, but that may take some time to hash out

**Pull request merged**